atomix/cluster/src/main/java/io/atomix/raft/storage/system/MetaStore.java

```diff
@@ -102,14 +102,16 @@ public MetaStore(final RaftStorage storage) throws IOException {
         FileChannel.open(confFile.toPath(), StandardOpenOption.READ, StandardOpenOption.WRITE);
 
     // Read existing meta info and rewrite with the current version
-    initializeMetaBuffer();
     lastFlushedIndex = readLastFlushedIndex();
+
+    initializeMetaBuffer();
```

Comment:
It was working before, but I think by mistake. So this doesn't fix anything, but it's safer to do so because that's what we intended.

So essentially, since we write the meta buffer every time we modify it, we want to have it initialized before changing any of its properties (to avoid overwriting untouched properties).

This worked because in the `initializeMetaBuffer`, we were "loading" the `lastFlushedIndex` (which returns the cached value, aka 0), and storing it (which does nothing if the value is the same as the cached value).

But clearly this is not what we wanted to do, so it's safer to read it before initializing the buffer.

To be honest, I think we could also remove this initialization, as we already read the whole buffer, but I wanted to keep the refactoring here to a minimum.